src/components/containers/Fold.js

```diff
@@ -63,9 +66,22 @@ export default class Fold extends Component {
       </div>
     );
 
-    const foldContent = !folded ? (
-      <div className={contentClass}>{children}</div>
-    ) : null;
+    let foldContent = null;
+    if (!folded && !isEmpty) {
+      foldContent = <div className={contentClass}>{children}</div>;
+    }
+
+    if (!folded && isEmpty) {
+      foldContent = (
+        <div className={contentClass}>
+          <FoldEmpty
+            icon={Icon}
+            messagePrimary={isEmpty.messagePrimary}
+            messageSecondary={isEmpty.messageSecondary}
```

Comment:
My only concern here is having two props for messages (What if we need three or more?), versus passing a message prop that could be either a single node or an array of strings/components, eg:

pseudo code --

```
const checkType = (message) => {
    if (typeof message === 'array') {
        message.map((messageItem, i) => <div key={i} className="message">{message}</div>)
    } else {
        return (<div className="message">{message}</div>);
    }
}

message={[_('localized string 1), _('localized string 2)]} || message={_('localized single string)} || message={<SingleComponent />}
```

Reply:
yeah, we could envision this as we extend the functionality, for now I think it'll work ok